lib/config-generator.js

```diff
@@ -290,6 +294,10 @@ class ConfigGenerator {
 
         if (this.webpackConfig.useSassLoader) {
             rules.push({
+                resolve: {
+                    mainFields: ['sass', 'style', 'main'],
+                    extensions: ['.scss', '.sass', '.css']
```

Comment:
What exactly does this do in the process? I know what `resolve.extensions` does at the root level - it allows you to omit the file extension. But what about here? If I say `import 'foo-bar'` where `foo-bar` is a module, what is the purpose of the `extensions`?

Reply:
The `resolve.extensions` value overrides the value at the root level. When you say `@import 'foo-bar'` Webpack will first check for an alias called `foo-bar`. If it does not find an alias for `foo-bar` it will check each directory specified in `resolve.modules` for a directory called `foo-bar`. Assuming it finds that directory, it will then look to see if there is a package.json file for the module. If there is, it will use the `resolve.mainFields` setting to determine which file to import. If there is not then it will use the `resolve.mainFiles` in conjunction with `resolve.extentions` to determine what file to look for in that directory.

If you were to say `@import './foo-bar'` then Webpack would look in the current directory for a file named foo-bar with an extension of `.scss`, `.sass`, or `.css` in that order.

Reply:
One more question: because this is being added to the sass loader, this resolution logic applies to files being imported from *within* a Sass file only, right? So, for example, if I were in `foo.css` (the earlier import`), `./foo-bar` would look for `foo-bar.css` but not `foo-bar.scss` (because that rule only lives down here).

What about less & stylus below? Do those also need these options?

Reply:
Yes, these are loader specific options. So `@import "./foo-bar"` in `foo.css` would only look for `./foo-bar.css` and not `foo-bar.scss`.

These options can be specified for any loaders where appropriate so long as that loader uses Webpack to perform the import resolution.